### DISCUSSION TITLE
How will I deploy an express server written in typescript?

### DISCUSSION THREAD

**Discussion by stephenasuncionDEV** [vercel/community#841] [Help] [answered]
I'm trying to figure out how will I deploy an express server written in typescript.
With my current configurations It deploys successfully. However, it is not running at all. 

--- Vercel dashboard config
Build Command: **tsc**
Output Directory: **.**

--- Project
Main File: **index.js**

package.json
```
"scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "dev": "nodemon index.ts",
    "start": "node index.js"
},
```
vercel.json
```
{
    "version": 2,
    "builds": [
        {
            "src": "./index.js",
            "use": "@vercel/node"
        }
    ]
}
```

Log:
![image](https://user-images.githubusercontent.com/73958774/189587062-9a8bef9f-bf47-4578-817b-2ad69958abe0.png)

Actual Website:
![image](https://user-images.githubusercontent.com/73958774/189587125-3fdba26b-b2e3-475f-a907-b50573da2c32.png)

**Comment by swarnava** [comment, marked as answer]
Hi,

You can  deploy APIs using pages/api in Next.js → https://nextjs.org/docs/api-routes/introduction.

See also:  https://vercel.com/guides/using-express-with-vercel

The general rule is not to deploy an entire, monolithic Node.js/Express.js server on a pages/api which is a Serverless Function/AWS Lambda underneath. Serverless Functions are not meant to take on the load of an entire Node.js server.

**Comment by stephenasuncionDEV** [comment]
Fixed! If you want to deploy your typescript express.js server on Vercel, here is my configuration:

vercel.json
```
{
    "version": 2,
    "builds": [{ "src": "index.ts", "use": "@vercel/node" }],
    "routes": [{ "src": "(.*)", "dest": "index.ts" }]
}
```

package.json scripts
```
"scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "dev": "nodemon index.ts",
    "start": "node index.js",
    "build": "tsc"
}
```
